cli/specifier_handler.rs

```diff
@@ -114,6 +114,17 @@ pub struct Dependency {
   pub maybe_type: Option<ModuleSpecifier>,
 }
```

Comment:
Is it ever the case that `maybe_code.is_some() && maybe_type.is_some()` ? What about the opposite - is it possible that `maybe_code.is_none() && maybe_type.is_none()` ?

If the answer to those questions is no, then consider refactoring this type to not allow for those situations. It might be something like this:
```rust
struct Dependency {
  location: Location,
  is_dynamic: bool,
  kind: DependencyKind,
  module_specifier: ModuleSpecifier,
}

enum DependencyKind { Code, Type };
```

Reply:
`maybe_code.is_some()` && `maybe_type.is_some()` occurs quite frequently when you have `@deno-types` or `X-TypeScript-Types` or `/// <reference types="" />`.  Modelling this properly is one of the key improvements in the new module graph, so we can always ensure we have a single dependency structure and not get weird edge cases.

`maybe_code.is_none()` && `maybe_type.is_none()` only occurs in situations situations where dynamic imports potentially have runtime issues, so we see the dependency in the graph, but we "defer" erroring until the module is expressly attempted to be loaded by `deno_core`.